zrml/prediction-markets/src/lib.rs

```diff
@@ -1385,6 +1389,10 @@ mod pallet {
                     },
                 );
 
+                total_weight = total_weight.saturating_add(open.unwrap_or_else(|_| {
+                    T::WeightInfo::market_status_manager(CacheSize::get(), CacheSize::get())
+                }));
```

Comment:
[nitpick^2] Same thing here - I'd either use the `unwrap_or_else` pattern everywhere or nowhere in this function.